src/components/views/dialogs/ReportEventDialog.tsx

```diff
@@ -0,0 +1,415 @@
+                // Report to moderators through to the dedicated bot,
+                // as configured in the room's state events.
+                const client = MatrixClientPeg.get();
+                const dmRoomId = await ensureDMExists(client, this.moderatedByUserId);
+                const ev = this.props.mxEvent;
+                await MatrixClientPeg.get().sendEvent(dmRoomId, ABUSE_EVENT_TYPE, {
+                    event_id: ev.getId(),
+                    room_id: ev.getRoomId(),
+                    moderated_by_id: this.moderatedByRoomId,
+                    nature: this.state.nature,
+                    reporter: MatrixClientPeg.get().getUserId(),
+                    comment: this.state.reason.trim(),
+                });
+                this.props.onFinished(true);
+            } else {
+                // Report to homeserver admin through the dedicated Matrix API.
+                const ev = this.props.mxEvent;
+                await MatrixClientPeg.get().reportEvent(ev.getRoomId(), ev.getId(), -100, this.state.reason.trim());
+                this.props.onFinished(true);
+            }
+        } catch (e) {
+            this.setState({
+                busy: false,
+                err: e.message,
+            });
+        }
+    };
+
+    render() {
+        const BaseDialog = sdk.getComponent('views.dialogs.BaseDialog');
+        const DialogButtons = sdk.getComponent('views.elements.DialogButtons');
+        const Loader = sdk.getComponent('elements.Spinner');
+        const Field = sdk.getComponent('elements.Field');
+
+        let error = null;
+        if (this.state.err) {
+            error = <div className="error">
+                {this.state.err}
+            </div>;
+        }
+
+        let progress = null;
+        if (this.state.busy) {
+            progress = (
+                <div className="progress">
+                    <Loader />
+                </div>
+            );
+        }
+
+        const adminMessageMD =
+            SdkConfig.get().reportEvent &&
+            SdkConfig.get().reportEvent.adminMessageMD;
+        let adminMessage;
+        if (adminMessageMD) {
+            const html = new Markdown(adminMessageMD).toHTML({ externalLinks: true });
+            adminMessage = <p dangerouslySetInnerHTML={{ __html: html }} />;
+        }
+
+        if (this.moderatedByRoomId && this.moderatedByUserId) {
+            // Display report-to-moderator dialog.
+            // We let the user pick a nature.
+            const client = MatrixClientPeg.get();
+            const homeServerName = SdkConfig.get()["validated_server_config"].hsName;
+            let subtitle;
+            switch (this.state.nature) {
+                case NATURE_DISAGREEMENT:
+                    subtitle = _t("Abuse: What this user is writing is wrong.");
+                    break;
+                case NATURE_TOXIC:
+                    subtitle = _t("Abuse: This user is displaying toxic behavior.");
```

Comment:
```suggestion
                    subtitle = _t("Abuse: This user is displaying toxic behaviour.");
```

Element is using british spelling by default